src/Resources/contao/widgets/PickerWidget.php

```diff
@@ -0,0 +1,250 @@
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao;
+
+
+/**
+ * Provide methods to handle input field "picker".
+ *
+ * @property boolean        $multiple
+ * @property array          $rootNodes
+ * @property string         $fieldType
+ * @property string         $foreignTable
+ * @property string         $context
+ * @property string         $labelField
+ * @property string         $icon
+ * @property boolean        $sortable
+ * @property array|callable $label_callback
+ *
+ * @author Leo Feyer <https://github.com/leofeyer>
+ * @author Kamil Kuzminski <https://github.com/qzminski>
+ */
+class PickerWidget extends \Widget
+{
+
+	/**
+	 * Submit user input
+	 * @var boolean
+	 */
+	protected $blnSubmitInput = true;
+
+	/**
+	 * Template
+	 * @var string
+	 */
+	protected $strTemplate = 'be_widget';
+
+
+	/**
+	 * Load the database object
+	 *
+	 * @param array $arrAttributes
+	 */
+	public function __construct($arrAttributes=null)
+	{
+		$this->import('Database');
+		parent::__construct($arrAttributes);
+	}
+
+
+	/**
+	 * Return an array if the "multiple" attribute is set
+	 *
+	 * @param mixed $varInput
+	 *
+	 * @return mixed
+	 */
+	protected function validator($varInput)
+	{
+		$this->checkValue($varInput);
+
+		if ($this->hasErrors())
+		{
+			return '';
+		}
+
+		// Return the value as usual
+		if ($varInput == '')
+		{
+			if ($this->mandatory)
+			{
+				$this->addError(sprintf($GLOBALS['TL_LANG']['ERR']['mandatory'], $this->strLabel));
+			}
+
+			return '';
+		}
+		elseif (strpos($varInput, ',') === false)
+		{
+			return $this->multiple ? array(intval($varInput)) : intval($varInput);
+		}
+		else
+		{
+			$arrValue = array_map('intval', array_filter(explode(',', $varInput)));
+
+			return $this->multiple ? $arrValue : $arrValue[0];
+		}
+	}
+
+
+	/**
+	 * Check the selected value
+	 *
+	 * @param mixed $varInput
+	 */
+	protected function checkValue($varInput)
+	{
+		if ($varInput == '' || !is_array($this->rootNodes) || !$this->Database->fieldExists('pid', $this->foreignTable))
+		{
+			return;
+		}
+
+		if (strpos($varInput, ',') === false)
+		{
+			$arrIds = array(intval($varInput));
+		}
+		else
+		{
+			$arrIds = array_map('intval', array_filter(explode(',', $varInput)));
+		}
+
+		if (count(array_diff($arrIds, array_merge($this->rootNodes, $this->Database->getChildRecords($this->rootNodes, $this->foreignTable)))) > 0)
+		{
+			$this->addError($GLOBALS['TL_LANG']['ERR']['invalidRecords']);
+		}
+	}
+
+
+	/**
+	 * Generate the record label
+	 *
+	 * @param array $data
+	 *
+	 * @return string
+	 */
+	protected function generateRecordLabel(array $data)
+	{
+		// Label callback
+		if (is_array($this->label_callback))
+		{
+			$callback = $this->label_callback;
+
+			$label = static::importStatic($callback[0])->{$callback[1]}($data, $this->objDca);
+		}
+		elseif (is_callable($callback = $this->label_callback))
+		{
+			$label = $callback($data, $this->objDca);
+		}
+		elseif ($this->labelField)
```

Comment:
Maybe check if data contains `labelField` ? I would also suggest to make the field mandatory (throw exception if it does not exist here).

Reply:
Changed in b853270. I have not used the `isset()` on purpose because the data property could potentially contain a `null` value.